doc/source/ray-core/tasks.rst

```diff
@@ -177,10 +177,24 @@ Multiple returns
 
       @ray.remote(num_returns=3)
       def return_multiple():
-          return 1, 2, 3
+          return 0, 1, 2
 
       a, b, c = return_multiple.remote()
 
+    For tasks that return multiple objects, Ray also supports remote generators that allow a task to return one object at a time to reduce memory usage at the worker. See the :ref:`user guide <generators>` for more details on use cases.
+
+    .. code-block:: python
+
+      @ray.remote(num_returns=3)
+      def return_multiple_as_generator():
+          for i in range(3):
+              yield i
+
+      # NOTE: Similar to normal functions, these objects will not be available
```

Comment:
Yeah, good point! For now, I think we should use the same semantics as normal tasks with multiple returns, especially since the ObjectRefs returned to the caller are a list, not a generator. But if we decide to support the generator interface in the future, I do think we should consider providing the iterator semantics instead.